src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,148 @@
+package duke.parser;
+
+import java.util.stream.Stream;
+
+import duke.DukeException;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.TodoCommand;
+import duke.command.UnknownCommand;
+import duke.command.UnmarkCommand;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+/**
+ * This class handles parsing of user commands to Command objects.
+ */
+public class Parser {
+
+    /**
+     * Parses the specified String parameter into a Command object.
+     * @param command the String to be parsed.
+     * @return Command.
+     * @throws DukeException if command is malformed.
+     */
+    public static Command parse(String command) throws DukeException {
+
+        Argument[] args = processArgs(command);
+        String commandName = args[0].getName();
+
+        switch (commandName) {

```

Comment:
I like your implementation of `switch` it makes the code look very clean. 👍

Reply:
Thanks!